cosmoz-treenode-navigator.js

```diff
@@ -162,108 +163,45 @@
 		/**
 		 * Returns a node array with the children of the given path.
 		 */
-		_renderLevel: function (pl, nodes) {
-			var pathSegment = nodes,
-				pathArray,
-				level = [];
-			if (!pl) {
-				// Return the formatted root nodes.
-				level = Object.keys(nodes).map(function (key) {
-					var node = nodes[key];
-					return {
-						id: key,
-						name: node[this.comparisonProperty],
-						path: node.pathLocator,
-						children: node[this.childProperty]
-					};
-				}, this);
-				return this._sortItOut(level);
+		_renderLevel: function (pathLocator, tree) {
+			if (!tree) {
+				return;
 			}
-
-			pathArray = pl.split(this.separatorSign);
-			pathArray.forEach(function (pathKey, i, arr) {
-				var node = pathSegment[pathKey],
-					children = node[this.childProperty];
-				if (i === arr.length - 1) {
-					level = Object.keys(children).map(function (childKey) {
-						var child = children[childKey];
-						return {
-							id: child.key,
-							name: child[this.comparisonProperty],
-							path: child.pathLocator,
-							children: child[this.childProperty]
-						};
-					}, this);
-				} else {
-					pathSegment = children;
-				}
-			}, this);
-
-			return this._sortItOut(level);
+			var node = tree.getNodeByPathLocator(pathLocator),
+				children = tree.getChildren(node),
+				level = children || node;
+			return this._sortNodes(this._normalizeNodes(level));
 		},
 		/**
-		 * Returns an Array of nodes on a given path.
+		 * Normalizes and returns an Array of nodes
+		 * with the properties name, path, sectionName, children
 		 */
-		_getNodesOnPath: function (pl, nodes) {
-			var path = pl.split(this.separatorSign),
-				pathSegment = nodes;
-				
-			return path.map(function (nodeKey) {
-				var node = pathSegment[nodeKey],
-					children = node[this.childProperty];
-				node['key'] = nodeKey;
-				if (node && children !== undefined && Object.keys(children).length > 0) {
-					pathSegment = children;
-				}
-				return node;
+		_normalizeNodes: function (nodes) { 
+			return nodes.map(function (node) {
+				var path = node.pathLocator || node.path;
+				return {
+					name: node[this.comparisonProperty],
```

Comment:
`this.comparisonProperty` should be replaced with `this.tree.searchProperty` wherever possible.
Same goes for `this.childProperty` -> `this.tree.childProperty`.